---
title: "Trap Reanalysis"
output:
  html_document:
    keep_md: yes
    theme: united
    toc: yes
  pdf_document:
    toc: yes
  word_document:
    toc: yes
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE,
  fig.path = "README_figs/README-"
)
```

required r libraries
```{r results='hide', message=FALSE, warning=FALSE}
library(edgeR)
library(SEtools)
library(SummarizedExperiment)
library(plgINS)
library(DESeq2)
library(sva)
```


# About this re-analysis

Here, we outline the re-analysis of two GEO datasets, GSE100579 and GSE131972, both of which are investigating the effect of acute stress on the translatome of CA3 pyramidal neurons in the mouse hippocampus.

This repositories were used for 3 seperate publications:

Marrocco J. et al. 2017; 
A sexually dimorphic pre-stressed translational signature in CA3 pyramidal neurons of BDNF Val66Met mice; Nature Communications volume 8, Article number: 808 (2017)

Gray J. D. et al. 2018; 
Translational profiling of stress-induced neuroplasticity in the CA3 pyramidal neurons of BDNF Val66Met mice; Molecular Psychiatry volume 23, pages 904â€“913

Marrocco J. et al. 2019; 
Early Life Stress Restricts Translational Reactivity in CA3 Neurons Associated With Altered Stress Responses in Adulthood; Front. Behav. Neurosci. https://doi.org/10.3389/fnbeh.2019.00157 


the reanalysis be performed on the quantification of two different alginment methods a genome alignment using xxx and salmon or a pseudoalignment to the transcriptome using kallisto

first, we are loading both the kallisto and the genome alignment data

```{r}
kallistodata <- readRDS("data/AllData.kallisto.SE.rds")
salmondata <- readRDS("data/GSE131972.salmonv2.SE.rds")
```

# Re-analysis of published results
## Assessment of Marrocco et al. 2017

In the 2017 publication the authors claim that numerous genes are differentially regulated between males and females after acute stress. Unfortunately, they do not include a list with all genes in the publication, but Table 1 contains a subsets of genes that are allegedly differentially regulated between males and females upon acute stress

First, we inspect these genes across the runs used in the original publication in WildType animals

```{r}
se <- kallistodata
se <- subset(se,select  = se$Set == "GSE100579" & se$Genotype == "WildType")
genes <- read.table("metadata/Marrocco2017SexStressGenes.csv", sep = ";", header = T)$genes
se <- se[,order(se$Sex,se$FST)]
sehm(se,genes,do.scale = T,anno_columns = c("FST","Sex"), cluster_rows = T, main = "Genes with male-female acute stress differences")
```

It looks as if there could be something to this, however unfortunately this finding is higly underpowered having only 6 replicates for a 4 group comparison

Let's inspect these genes across all sequencing runs and find out wheather the finding for male WildType mice in the second set GSE131972 holds up.

```{r}
se <- kallistodata
se <- subset(se,select  = se$ELS == "None" & se$Genotype == "WildType")
se <- se[,order(se$Set,se$Sex,se$FST)]
sehm(se,genes,do.scale = T,anno_columns = c("FST","Sex","Set"), gaps_at = "Set", cluster_rows = T, main =  "Same genes in the second data-set")
```

It becomes apparent that these genes seem to be co-expressed and highly variable across all samples. the finding that in females and males these genes react differently to stress seems inaccurate, with replicates from the second set demonstrating strong intra-group differences.

If we further visualize these genes across all samples, we find
```{r}
se <- kallistodata
se <- se[,order(se$Sex,se$FST)]
sehm(se,genes,do.scale = T,anno_columns = c("ELS","FST","Genotype","Sex","Set"), cluster_rows = T, gaps_at = "Set",   main =  "Same genes across all samples")
```

The selected genes display a high co-expression and could possibly be highly variable due to technical reasons. Let's try to remove technical variability from our runs using SVA and re-visualize the same genes

```{r}
#function for SVA correction
dosvacor <- function(SE, form=NULL, form0=~1, ...){
  CD <- as.data.frame(colData(SE))
  mm <- model.matrix(form, data=CD)
  mm0 <- model.matrix(form0, data=CD)

  dds <- DESeqDataSetFromMatrix(round(assay(SE)), as.data.frame(colData(SE)), form)
  dds <- estimateSizeFactors(dds)
  en <- as.matrix(assay(vst(dds, blind=FALSE)))

  sv <- sva(en, mm, mm0, n.sv=NULL, ...)
  n.sv <- sv$n.sv
  sv <- sv$sv
  
  colnames(sv) <- paste0("SV",1:ncol(sv))
  X <- cbind(mm, sv)
  mm2 <- cbind(mm[,1,drop=F],sv,mm[,-1,drop=F])
  H <- solve(t(X)%*%X)%*%t(X)
  b <- (H%*%t(en))
  cn <- setdiff(colnames(X),setdiff(colnames(mm), colnames(mm0)))  
  cn <- setdiff(cn, "(Intercept)")
  encor <- en - t(as.matrix(X[,cn]) %*% b[cn,])
  SE <- SE[row.names(encor),]
  colData(SE) <- cbind(colData(SE), sv)
  assays(SE)$corrected <- encor
  return(SE)
}

se <- dosvacor(se, form = ~ Set + Sex * FST, form0 = ~Set)
sehm(se,genes,do.scale = T,anno_columns =  c("ELS","FST","Genotype","Sex","Set"), cluster_rows = T,assayName = "corrected",  main =  "Same genes after correcting for technical variability")
```

As we can see, removing technical variability abolishes the effects for the male-female stress difference seen in the single replicate comparisons. It becomes apparent, that these genes do not have any sex specific acute stress response across all samples

## Assessment of Gray et al. 2018

A complete assessment of the findings of Gray et al. 2018 is unfortunately not possible. the repository GSE100579 is missing crucial samples for the chronic stress model for the BDNF Val66Met genotype and only includes samples of acute stress.

However, the authors claim that many genes are differentially regulated between WildType and BDNF Val66Met animals at baseline. While no complete list of genes has been included in the publication, a subset can be found in Table 1.

We plot these genes in the GSE100579 data set.
```{r}
se <- kallistodata
genes <- read.table("metadata/Gray2018GenotypeGenes.csv", sep = ";", header = T)$genes
se <- subset(se,select  = se$Set == "GSE100579" & se$FST == "None")
se <- se[,order(se$Sex,se$Genotype)]
sehm(se,genes,do.scale = T,anno_columns = c("Genotype","Sex"), cluster_rows = T, main = "Genes differentially effected by BDNF Val66Met in males and females")
```

Again, we find that the experimental design was severly underpowered.
We investigate the expression of these genes across all samples from both datasets.
```{r}
se <- kallistodata
se <- se[,order(se$Sex,se$Genotype)]
sehm(se,genes,do.scale = T,anno_columns = c("ELS","FST","Genotype","Sex","Set"), cluster_rows = T, gaps_at = "Set", main = "Same genes across all samples")
```

We see a similar pattern as in Marrocco et al 2017. The baseline difference does not reproduce in GSE 131972 which only includes male WildType mice and shows high inter-group variability for these genes.

Let's again try to eliminate technical variabilty and re-visualize the same genes

```{r}
se <- dosvacor(se, form = ~Set + Sex + Genotype + FST + ELS, form0 = ~Set)
sehm(se,genes,do.scale = T,anno_columns = c("ELS","FST","Genotype","Sex","Set"), cluster_rows = T,assayName = "corrected", main = "Same genes after removing technical variabilty")
```

It becomes apparent, that these genes do not have any genotype specific baseline difference across all samples

## Assessment of Marrocco et al. 2019
Here we re-analyse Marrocco et al. 2019 checking for ELS dependent changes in the acute stress response. The same analysis will be performed three times, once using gene counts from a transcriptome alignment (kallisto) with edgeR, once using gene counts from a genome alignment with edgeR and once using gene counts from a transcriptome alignment (kallisto) with DESeq. All analyses had similar results, whereas DESeq produced most significant hits. However, the core message of the publication that ELS substantially impacts the transcriptional acute stress response could not be reproduced with any method.

### Visualization of genes mentioned in the publication 

```{r}
se <- kallistodata
se <- se[order(rownames(se)),]
se <- subset(se,select  = se$Set == "GSE131972")
se2 <- salmondata
se2 <- se2[order(rownames(se2)),]
```

In their publications the authors unfortunately do not upload a list with differentially expressed genes. However, in their discussion they mention a number of genes that they claim are differentially expressed between ELS and non-ELS mice after acute stress.
They claim that acute stress reduces the expression of Grin1, Grin2a, Gabbr2, and Gabra1 in CA3 neurons of non-ELS mice, but not ELS mice:
```{r}
sehm(se,c("Grin1","Grin2a","Gabbr2","Gabra1"),do.scale = T ,anno_columns = c("ELS","FST"),  main = "Selected genes with transcriptome alignment")
sehm(se2,c("Grin1","Grin2a","Gabbr2","Gabra1"),do.scale = T,anno_columns = c("ELS","FST"), , assayName = "TPM",  main = "Selected genes with genome alignment")
```

it becomes apparent, that this finding does not look robust

Further, they claim that there is restricted list of genes selectively induced by AS in ELS mice (Per1, Npy, Nfkbia, Penk,Dusp1, Cst3, Trib1, Htra1, Sdc4, Plekhf1) but not non-ELS mice.
```{r}
sehm(se,c("Per1", "Npy", "Nfkbia", "Penk","Dusp1", "Cst3", "Trib1", "Htra1", "Sdc4", "Plekhf1"),do.scale = T,anno_columns = c("ELS","FST"),  main = "Selected genes with transcriptome alignment")
sehm(se2,c("Per1", "Npy", "Nfkbia", "Penk","Dusp1", "Cst3", "Trib1", "Htra1", "Sdc4", "Plekhf1"),do.scale = T,anno_columns = c("ELS","FST"), assayName = "TPM",  main = "Selected genes with genome alignment")
```

it becomes apparent, that this finding also does not look robust while these genes might be increased in expression in ELS mice it is hard to see this being a significant finding, especially, since the same effect can be observed for one replicate in the non-ELS group.

The authors claim that there are a number of genes that appear to be induced by AS in both ELS and non ELS mice. these include (Egr1/2/4, Arc, Fos, and Fosb)
```{r}
sehm(se,c("Egr1", "Egr2", "Egr4", "Arc","Fos", "Fosb"),do.scale = T,anno_columns = c("ELS","FST"), main = "Selected genes using transcriptome alignment")
sehm(se2,c("Egr1", "Egr2", "Egr4", "Arc","Fos", "Fosb"),do.scale = T,anno_columns = c("ELS","FST"), assayName = "TPM", main = "Selected genes using genome alignment")
```

For these genes it looks as if the finding might be significant.

### Analysis using kallisto data and edgeR

Let's investigate the transcriptome alligned reads with a statistical approach using egeR and a GLM type of analysis for swim effects, early life effects and interactions
```{r}
#experimental design, interactive model
design <- model.matrix(~se$FST * se$ELS)

y <- DGEList(counts=assays(se)$counts)
y <- calcNormFactors(y)
y <- estimateDisp(y,design)

#filter out genes that are below 10 counts in more than 75% of samples
keep <- rowSums(y$counts>10) >= (ncol(y) * 0.25)
y <- y[keep, , keep.lib.sizes=FALSE]

Results <- list()
fit <- glmQLFit(y,design)
for(i in colnames(design)[-1]){
  Results[[i]] <- glmQLFTest(fit, i)
}
```

Les's investigate if there are any genes altered by acute stress
```{r}
topTags(Results$`se$FSTFST`)
```

Even though the data looked promising no genes pass mutliple testing correction. This indicates that this study might be severly underpowered, so in order to resolve FST specific effects more replicats would be needed

Are there any genes altered by early life stress?
```{r}
topTags(Results$`se$ELSELS`)
```

There are no genes that pass the multiple testing correction.

Let's investigate if there are genes with a significant interaction
```{r}
topTags(Results$`se$FSTFST:se$ELSELS`)
```

No genes have a altered acute stress response in ELS vs normal animals. This clearly fails to reproduce the findings of the original study, which claimed hundereds of genes being altered in their FST respons following ELS

### Using salmon data and edgeR

Let's investigate the genome alligned counts with a statistical approach using egeR and a GLM type of analysis for swim effects, early life effects and interactions
```{r}
#experimental design, interactive model
design <- model.matrix(~se$FST * se$ELS)

y <- DGEList(counts=assays(se)$counts)
y <- calcNormFactors(y)
y <- estimateDisp(y,design)

#filter out genes that are below 10 counts in more than 75% of samples
keep <- rowSums(y$counts>10) >= (ncol(y) * 0.25)
y <- y[keep, , keep.lib.sizes=FALSE]

Results <- list()
fit <- glmQLFit(y,design)
for(i in colnames(design)[-1]){
  Results[[i]] <- glmQLFTest(fit, i)
}
```

Les's investigate if there are any genes altered by acute stress
```{r}
topTags(Results$`se$FSTFST`)
```

No genes pass mutliple testing correction

Are there any genes altered by early life stress?
```{r}
topTags(Results$`se$ELSELS`)
```

There are no genes that pass the multiple testing correction

Let's investigate if there are genes with a significant interaction
```{r}
topTags(Results$`se$FSTFST:se$ELSELS`)
```

There are no genes have an altered acute stress response in ELS vs normal animals

### Using DESeq instead of EdgeR for statistical analysis

The authors of the orignial publication used DESeq for the statistical assessment. For completeness, we repeat the same analysis using this method on the transcriptome alligned data

```{r}
suppressPackageStartupMessages({
  library(DESeq2)
  library(sva)
  library(SEtools)
  library(RUVSeq)
})
```


```{r}
dat1 <- kallistodata
dat1 <- dat1[,dat1$Set=="GSE131972"]
dds1 <- DESeqDataSetFromMatrix(round(assay(dat1)), colData=colData(dat1),
        design=~FST+ELS+FST:ELS )
dds1 <- DESeq(dds1)
# so that we plot the DESeq-normalized counts:
dat1 <- dat1[row.names(dds1),]
assays(dat1)$lognorm <- log1p(counts(dds1, normalized=TRUE))
rowData(dat1)$logCount <- rowMeans(assays(dat1)$lognorm)
resultsNames(dds1)
```


```{r}
res <- results(dds1, name = "FST_FST_vs_None")
DESeq2::plotMA(res)
res <- res[order(res$padj),]
dat1 <- dat1[,order(dat1$ELS, dat1$FST)]
sehm(dat1, row.names(res)[which(res$padj<0.05)], assayName = "lognorm", do.scale = TRUE, anno_row="logCount", anno_columns=c("ELS","FST"), main="Differentially expressed upon FST")
```

We find a few genes which are differentially expressed after FST


```{r}
res <- results(dds1, name = "ELS_ELS_vs_None")
DESeq2::plotMA(res)
res <- res[order(res$padj),]
dat1 <- dat1[,order(dat1$ELS, dat1$FST)]
sehm(dat1, row.names(res)[which(res$padj<0.05)], assayName = "lognorm", do.scale = TRUE, anno_row="logCount", anno_columns=c("ELS","FST"), main="Differentially expressed upon ELS")
```

We find a few genes which are differentially expressed after ELS


```{r}
res <- results(dds1, name = "FSTFST.ELSELS")
DESeq2::plotMA(res)
res <- res[order(res$padj),]
dat1 <- dat1[,order(dat1$ELS, dat1$FST)]
res[which(res$padj<0.05),]
```
We find a single gene that is differentially expressed after FST in ELS vs non-ELS mice. Still, this clearly fails to reproduce the findings of the original study, which claimed thousands of genes being altered in their FST respons following ELS

# Meta-Analysis of all data
Since we were unable to reproduce many of the findigs from the original publications we re-assess the combined data from both repositiories to investigate a number of questions:

1. are genes differentially expressed following forces swim stress?
2. are genes differentially expressed following males and females?
3. are genes differentially expressed in BDNF Val66Met mice?
4. are genes differentially expressed following early life stress
5. are responses of FST genes altered by sex?
6. are responses of FST genes altered by BDNF Val66Met?
7. are responses of FST genes altered by early life stress?

## Additive model
Let's run an overarching analysis over all data to determine if there are any significant effects for forces swim stress (=FST), Sex, Genotype or early life stress (=ELS). In the process we also remove technical variabilty to increase the chance of successfully find candidate genes.
```{r}
se <- kallistodata
se <- dosvacor(se, form = ~FST + Sex + Genotype + Set + ELS, form0 = ~Set)

#experimental design, full additive model
design <- model.matrix(~ se$SV1 + se$SV2 + se$SV3 + se$FST + se$Sex + se$Genotype + se$Set + se$ELS)

y <- DGEList(counts=assays(se)$counts)
y <- calcNormFactors(y)
y <- estimateDisp(y,design)

#filter out genes that are below 10 counts in more than 75% of samples
keep <- rowSums(y$counts>10) >= (ncol(y) * 0.25)
y <- y[keep, , keep.lib.sizes=FALSE]

Results <- list()
fit <- glmQLFit(y,design)
for(i in colnames(design)[-1]){
  Results[[i]] <- glmQLFTest(fit, i)
}
```

### Are genes differentially expressed following forces swim stress?
```{r}
se <- se[,order(se$FST)]
sehm(se, rownames(topTags(Results$`se$FSTFST`,p.value = 0.05, n = 1000)), assayName = "corrected", do.scale = TRUE, anno_columns=c("ELS","FST","Genotype","Sex","Set"), main="Significant FST genes with corrected data")
sehm(se, rownames(topTags(Results$`se$FSTFST`,p.value = 0.05, n = 1000)), do.scale = TRUE, anno_columns=c("ELS","FST","Genotype","Sex","Set"), main="Significant FST genes with uncorrected data")
topTags(Results$`se$FSTFST`,p.value = 0.05, n = 1000)
```

There are multiple candidate genes that are significantly altered by acute stress across other condition

### Are genes differentially expressed between males and females?
```{r}
se <- se[,order(se$Sex)]
sehm(se, rownames(topTags(Results$`se$Sexfemale`,p.value = 0.05, n = 1000)), assayName = "corrected", do.scale = TRUE, anno_columns=c("ELS","FST","Genotype","Sex","Set"), main="Significant sex genes with corrected data")
sehm(se, rownames(topTags(Results$`se$Sexfemale`,p.value = 0.05, n = 1000)), do.scale = TRUE, anno_columns=c("ELS","FST","Genotype","Sex","Set"), main="Significant Sex genes with uncorrected data")
topTags(Results$`se$FSTFST`,p.value = 0.05, n = 1000)
```

There are multiple candidate genes that are significantly altered by sex across other conditions


### Are genes differentially expressed in BDNF Val66Met mice?
```{r}
se <- se[,order(se$Genotype)]
sehm(se, rownames(topTags(Results$`se$GenotypeBDNFMET`)), assayName = "corrected", do.scale = TRUE, anno_columns=c("ELS","FST","Genotype","Sex","Set"), main="Top10 Genotype genes (not significant!)")
topTags(Results$`se$GenotypeBDNFMET`)
```

No genes pass multiple testing correction

### Are genes differentially expressed following early life stress?
```{r}
se <- se[,order(se$ELS)]
sehm(se, rownames(topTags(Results$`se$ELSELS`)), assayName = "corrected", do.scale = TRUE, anno_columns=c("ELS","FST","Genotype","Sex","Set"), main="Top10 ELS genes (not significant!)")
topTags(Results$`se$ELSELS`)
```

No genes pass multiple testing correction

## Interactive models

To futher investigate if across all samples there is any variable that interacts with acute stress we will run a series of models that incorporate an interaction term between acute stress and any of the variables

### Are responses of FST genes altered by sex?
```{r}
se <- kallistodata
se <- dosvacor(se, form = ~FST + Sex + Genotype + Set + ELS + FST:Sex, form0 = ~Set)

#experimental design, full additive model
design <- model.matrix(~ se$SV1 + se$SV2 + se$FST + se$Sex + se$Genotype + se$Set + se$ELS + se$FST:se$Sex)

y <- DGEList(counts=assays(se)$counts)
y <- calcNormFactors(y)
y <- estimateDisp(y,design)

#filter out genes that are below 10 counts in more than 75% of samples
keep <- rowSums(y$counts>10) >= (ncol(y) * 0.25)
y <- y[keep, , keep.lib.sizes=FALSE]

Results <- list()
fit <- glmQLFit(y,design)
for(i in colnames(design)[-1]){
  Results[[i]] <- glmQLFTest(fit, i)
}

se <- se[,order(se$Sex,se$FST)]
sehm(se, rownames(topTags(Results$`se$FSTFST:se$Sexfemale`)), assayName = "corrected", do.scale = TRUE, anno_columns=c("FST","Sex"), main="Top10 Sex:FST interaction genes (not significant!)")
topTags(Results$`se$FSTFST:se$Sexfemale`)
```

No genes pass multiple testing correction

### Are responses of FST genes altered by BDNF Val66Met?
```{r}
se <- kallistodata
se <- dosvacor(se, form = ~FST + Sex + Genotype + Set + ELS + FST:Genotype, form0 = ~Set)

#experimental design, full additive model
design <- model.matrix(~se$SV1 + se$SV2 + se$FST + se$Sex + se$Genotype + se$Set + se$ELS + se$FST:se$Genotype)

y <- DGEList(counts=assays(se)$counts)
y <- calcNormFactors(y)
y <- estimateDisp(y,design)

#filter out genes that are below 10 counts in more than 75% of samples
keep <- rowSums(y$counts>10) >= (ncol(y) * 0.25)
y <- y[keep, , keep.lib.sizes=FALSE]

Results <- list()
fit <- glmQLFit(y,design)
for(i in colnames(design)[-1]){
  Results[[i]] <- glmQLFTest(fit, i)
}
se <- se[,order(se$Genotype,se$FST)]
sehm(se, rownames(topTags(Results$`se$FSTFST:se$GenotypeBDNFMET`, p.value = 0.05, n = 1000)), assayName = "corrected", do.scale = TRUE, anno_columns=c("FST","Genotype"), main="Significant FST:Genotype interaction genes with corrected data")
sehm(se, rownames(topTags(Results$`se$FSTFST:se$GenotypeBDNFMET`, p.value = 0.05, n = 1000)), do.scale = TRUE, anno_columns=c("FST","Genotype"), main="Significant FST:Genotype interaction genes with uncorrected data")
topTags(Results$`se$FSTFST:se$GenotypeBDNFMET`, p.value = 0.05, n = 20)
```

There are multiple genes that pass multiple testing correction. However, the data for these still looks noisy and genes that show up are highly co-expressed. The low replicate number might have impaired the ability to remove technical variability sufficiently for the Genotype:FST assessment, so the results should be interpreted with caution and more replicates would be needed.

### are responses of FST genes altered by early life stress?
```{r}
se <- kallistodata
se <- dosvacor(se, form = ~FST + Sex + Genotype + Set + ELS + FST:ELS, form0 = ~Set)


#experimental design, full additive model
design <- model.matrix(~se$SV1 + se$SV2 + se$FST + se$Sex + se$Genotype + se$Set + se$ELS + se$FST:se$ELS)

y <- DGEList(counts=assays(se)$counts)
y <- calcNormFactors(y)
y <- estimateDisp(y,design)

#filter out genes that are below 10 counts in more than 75% of samples
keep <- rowSums(y$counts>10) >= (ncol(y) * 0.25)
y <- y[keep, , keep.lib.sizes=FALSE]

Results <- list()
fit <- glmQLFit(y,design)
for(i in colnames(design)[-1]){
  Results[[i]] <- glmQLFTest(fit, i)
}

se <- se[,order(se$ELS,se$FST)]
sehm(se, rownames(topTags(Results$`se$FSTFST:se$ELSELS`)), assayName = "corrected", do.scale = TRUE, anno_columns=c("ELS","FST"), main="Significant FST:ELS interaction genes")
topTags(Results$`se$FSTFST:se$ELSELS`)
```

No genes pass multiple testing correction
